cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase11/HBaseQueueRegionObserver.java

```diff
@@ -344,19 +346,21 @@ public boolean next(List<Cell> results, ScannerContext scannerContext) throws IO
           consumerConfig = getConfigCache(env).getConsumerConfig(currentQueue);
         }
 
-        if (consumerConfig == null) {
-          // no config is present yet, so cannot evict
-          return hasNext;
-        }
-
         invalidTxData = false;
         if (state != null) {
-          long txId = QueueEntryRow.getWritePointer(cell.getRowArray(), currentQueueRowPrefix.length);
+          // TODO: figure out how to get rowKey for Salted Queues
+          byte[] rowKey = ShardedHBaseQueueStrategy.ROW_KEY_CONVERTER.apply(CellUtil.cloneRow(cell));
```

Comment:
@chtyim Looks like there are some header bytes in the row key before the queue row prefix. We tried using `QueueEntryRow.getWritePointer(cell.getRowArray(), cell.getRowOffset() + currentQueueRowPrefix.length);` yesterday and we got an incorrect transaction id for the row. 

Row key converter is being used to remove the header bytes. If the header length is constant then we can do `QueueEntryRow.getWritePointer(cell.getRowArray(), cell.getRowOffset() + header.length + currentQueueRowPrefix.length);`. Is the row key format documented somewhere?